From 92d0fba77d2cf2edacd8731afdb058f4b6b79e47 Mon Sep 17 00:00:00 2001
From: taodong <taodong@cvte.com>
Date: Sat, 22 Feb 2020 11:41:44 +0800
Subject: [PATCH] [merge][cp-from@1911cf82][bugfix][JIRA#SMARTPDT-3819][wifi][common]p2p not support persistent group

[what]p2p connect failed due to persistent group
[why]p2p nogotiate failed due save previous group
[how]p2p nogotiate show device not support persistent group

Change-Id: I03b510c44dbde147c143ab83eeebde386ffe6414
---

diff --git a/wpa_supplicant_8/src/p2p/p2p_build.c b/wpa_supplicant_8/src/p2p/p2p_build.c
index 2882c6a..8092204 100644
--- a/wpa_supplicant_8/src/p2p/p2p_build.c
+++ b/wpa_supplicant_8/src/p2p/p2p_build.c
@@ -62,6 +62,11 @@
 void p2p_buf_add_capability(struct wpabuf *buf, u8 dev_capab, u8 group_capab)
 {
 	/* P2P Capability */
+        //CVTE ADD for p2p group not support persistent feature.
+        wpa_printf(MSG_DEBUG, "P2P: not support persisitent group and modify %u to 0.", group_capab);
+	u8 group_flag = 0xdd;
+        group_capab = group_capab & group_flag;
+        //CVTE END
 	wpabuf_put_u8(buf, P2P_ATTR_CAPABILITY);
 	wpabuf_put_le16(buf, 2);
 	wpabuf_put_u8(buf, dev_capab); /* Device Capabilities */
diff --git a/wpa_supplicant_8/wpa_supplicant/p2p_supplicant.c b/wpa_supplicant_8/wpa_supplicant/p2p_supplicant.c
index 0eccd52..327d37c 100644
--- a/wpa_supplicant_8/wpa_supplicant/p2p_supplicant.c
+++ b/wpa_supplicant_8/wpa_supplicant/p2p_supplicant.c
@@ -1356,11 +1356,15 @@
 			persistent = ssid->p2p_persistent_group;
 			os_memcpy(go_dev_addr, wpa_s->global->p2p_dev_addr,
 				  ETH_ALEN);
-		} else
-			persistent = wpas_p2p_persistent_group(wpa_s,
-							       go_dev_addr,
-							       ssid->ssid,
-							       ssid->ssid_len);
+		} else {
+			//CVTE MODITY for remove persist group when create p2p group
+			//persistent = wpas_p2p_persistent_group(wpa_s,
+			//				       go_dev_addr,
+			//				       ssid->ssid,
+			//				       ssid->ssid_len);
+			wpa_printf(MSG_DEBUG, "p2p_gc persistent is %d",persistent);
+			//CVTE EDN
+		}
 	} else {
 		client = wpa_s->p2p_group_interface ==
 			P2P_GROUP_INTERFACE_CLIENT;
@@ -7013,8 +7017,12 @@
 	os_memset(go_dev_addr, 0, ETH_ALEN);
 	if (ssid->bssid_set)
 		os_memcpy(go_dev_addr, ssid->bssid, ETH_ALEN);
-	persistent = wpas_p2p_persistent_group(wpa_s, go_dev_addr, ssid->ssid,
-					       ssid->ssid_len);
+	//CVTE MODIFY for remove p2p persistent group feature
+	//persistent = wpas_p2p_persistent_group(wpa_s, go_dev_addr, ssid->ssid,
+	//				       ssid->ssid_len);
+	persistent = 0;
+	wpa_dbg(wpa_s, MSG_DEBUG, "wpas_p2p_completed froce p2p persisitent is 0.");
+	//CVTE END
 	os_memcpy(wpa_s->go_dev_addr, go_dev_addr, ETH_ALEN);
 
 	if (wpa_s->global->p2p_group_formation == wpa_s)
